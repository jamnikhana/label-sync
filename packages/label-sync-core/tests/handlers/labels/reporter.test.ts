import {
  LabelSyncReport,
  createTerminalReport,
} from '../../../src/handlers/labels'

describe('label sync reporter', () => {
  test('generates correct empty report', async () => {
    const report: LabelSyncReport = {
      config: {
        labels: {},
        strict: false,
      },
      options: { dryRun: true },
      repository: {
        name: 'label-sync',
        owner: {
          login: 'maticzav',
        },
        full_name: 'maticzav/label-sync',
      },
      additions: [],
      updates: [],
      removals: [],
    }

    const message = createTerminalReport(report)

    expect(message).toMatchSnapshot()
  })

  test('correctly generates empty additions report', async () => {
    const report: LabelSyncReport = {
      config: {
        labels: {
          test: {
            description: 'Testing sync.',
            color: '#123456',
          },
        },
        strict: true,
      },
      options: { dryRun: false },
      repository: {
        name: 'label-sync',
        owner: {
          login: 'maticzav',
        },
        full_name: 'maticzav/label-sync',
      },
      additions: [],
      updates: [
        { color: 'f266f4', default: false, description: '', name: 'basic' },
      ],
      removals: [
        { color: 'f266f4', default: false, description: '', name: 'basic' },
      ],
    }

    expect(createTerminalReport(report)).toMatchSnapshot()
  })

  test('correctly generates empty updates report', async () => {
    const report: LabelSyncReport = {
      config: {
        labels: {
          test: {
            description: 'Testing sync.',
            color: '#123456',
          },
        },
        strict: true,
      },
      options: { dryRun: false },
      repository: {
        name: 'label-sync',
        owner: {
          login: 'maticzav',
        },
        full_name: 'maticzav/label-sync',
      },
      additions: [
        { color: 'f266f4', default: false, description: '', name: 'basic' },
      ],
      updates: [],
      removals: [
        { color: 'f266f4', default: false, description: '', name: 'basic' },
      ],
    }

    expect(createTerminalReport(report)).toMatchSnapshot()
  })

  test('correctly generates empty removals report', async () => {
    const report: LabelSyncReport = {
      config: {
        labels: {
          test: {
            description: 'Testing sync.',
            color: '#123456',
          },
        },
        strict: true,
      },
      options: { dryRun: false },
      repository: {
        name: 'label-sync',
        owner: {
          login: 'maticzav',
        },
        full_name: 'maticzav/label-sync',
      },
      additions: [
        { color: 'f266f4', default: false, description: '', name: 'basic' },
      ],
      updates: [
        { color: 'f266f4', default: false, description: '', name: 'basic' },
      ],
      removals: [],
    }

    expect(createTerminalReport(report)).toMatchSnapshot()
  })

  test('correctly generates report', async () => {
    const report: LabelSyncReport = {
      config: {
        labels: {
          test: {
            description: 'Testing sync.',
            color: '#123456',
          },
        },
        strict: true,
      },
      options: { dryRun: false },
      repository: {
        name: 'label-sync',
        owner: {
          login: 'maticzav',
        },
        full_name: 'maticzav/label-sync',
      },
      additions: [
        {
          color: '#123456',
          description: 'Testing sync.',
          name: 'test',
          default: false,
        },
      ],
      updates: [],
      removals: [
        {
          color: '333333',
          default: false,
          description: '',
          name: 'bug/no-reproduction',
        },
        { color: 'f266f4', default: false, description: '', name: 'basic' },
        { color: '333333', default: false, description: '', name: 'kind/bug' },
      ],
    }

    expect(createTerminalReport(report)).toMatchSnapshot()
  })
})
