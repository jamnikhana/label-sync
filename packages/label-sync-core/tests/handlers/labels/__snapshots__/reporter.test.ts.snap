// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label sync reporter correctly generates report 1`] = `
"This is an autogenerated report for your project.
(dry run: "false")

New Labels:
- test
  Testing sync.

Updated Labels:
No updates.

Removed Labels:
- bug/no-reproduction
- basic
- kind/bug

"
`;

exports[`label sync reporter generates correct empty report 1`] = `
"This is an autogenerated report for your project.
(dry run: "true")

New Labels:
No new labels.

Updated Labels:
No updates.

Removed Labels:
No deadly changes.

We haven't removed any label yet.
If you want us to remove them, set strict to true."
`;
